ode/unit_test/Test_ODE_BDF.hpp

```diff
@@ -0,0 +1,830 @@
+//                     vec_type, mv_type, mat_type, scalar_type>
+//       solve_wrapper(mySys, t_start, t_end, 1000, y0, y_new, rhs, update,
+//       scale, y_vecs,
+//                     kstack, temp, jac);
+//   Kokkos::parallel_for(myPolicy, solve_wrapper);
+
+//   Kokkos::fence();
+// }
+
+template <class mat_type, class scalar_type>
+void compute_coeffs(const int order, const scalar_type factor,
+                    const mat_type& coeffs) {
+  std::cout << "compute_coeffs" << std::endl;
+
+  coeffs(0, 0) = 1.0;
+  for (int colIdx = 0; colIdx < order; ++colIdx) {
+    coeffs(0, colIdx + 1) = 1.0;
+    for (int rowIdx = 0; rowIdx < order; ++rowIdx) {
+      coeffs(rowIdx + 1, colIdx + 1) =
+          ((rowIdx - factor * (colIdx + 1.0)) / (rowIdx + 1.0)) *
+          coeffs(rowIdx, colIdx + 1);
+    }
+  }
+}
+
+template <class mat_type, class scalar_type>
+void update_D(const int order, const scalar_type factor, const mat_type& coeffs,
+              const mat_type& tempD, const mat_type& D) {
+  auto subD =
+      Kokkos::subview(D, Kokkos::pair<int, int>(0, order + 1), Kokkos::ALL);
+  auto subTempD =
+      Kokkos::subview(tempD, Kokkos::pair<int, int>(0, order + 1), Kokkos::ALL);
+
+  compute_coeffs(order, factor, coeffs);
+  auto R = Kokkos::subview(coeffs, Kokkos::pair<int, int>(0, order + 1),
+                           Kokkos::pair<int, int>(0, order + 1));
+  std::cout << "SerialGemm" << std::endl;
+  KokkosBatched::SerialGemm<
+      KokkosBatched::Trans::Transpose, KokkosBatched::Trans::NoTranspose,
+      KokkosBatched::Algo::Gemm::Blocked>::invoke(1.0, R, subD, 0.0, subTempD);
+
+  compute_coeffs(order, 1.0, coeffs);
+  auto U = Kokkos::subview(coeffs, Kokkos::pair<int, int>(0, order + 1),
+                           Kokkos::pair<int, int>(0, order + 1));
+  std::cout << "SerialGemm" << std::endl;
+  KokkosBatched::SerialGemm<
+      KokkosBatched::Trans::Transpose, KokkosBatched::Trans::NoTranspose,
+      KokkosBatched::Algo::Gemm::Blocked>::invoke(1.0, U, subTempD, 0.0, subD);
+}
+
+template <class device_type, class scalar_type>
+void test_Nordsieck() {
+  using execution_space = Kokkos::HostSpace;
+  StiffChemistry mySys{};
+
+  Kokkos::View<scalar_type**, execution_space> R("coeffs", 6, 6),
+      U("coeffs", 6, 6);
+  Kokkos::View<scalar_type**, execution_space> D("D", 8, mySys.neqs),
+      tempD("tmp", 8, mySys.neqs);
+  int order     = 1;
+  double factor = 0.8;
+
+  constexpr double t_start = 0.0, t_end = 500.0;
+  int max_steps = 200000;
+  double dt     = (t_end - t_start) / max_steps;
+
+  auto y0 = Kokkos::subview(D, 0, Kokkos::ALL());
+  auto f  = Kokkos::subview(D, 1, Kokkos::ALL());
+  y0(0)   = 1.0;
+
+  mySys.evaluate_function(0, 0, y0, f);
+  for (int eqIdx = 0; eqIdx < mySys.neqs; ++eqIdx) {
+    f(eqIdx) *= dt;
+  }
+
+  compute_coeffs(order, factor, R);
+  compute_coeffs(order, 1.0, U);
+
+  {
+    std::cout << "R: " << std::endl;
```

Comment:
Is this meant to be permanent? It seems like the kind of thing that should be printed only if a test fails